plugin.video.hardstyletv/plugintools.py

```diff
@@ -0,0 +1,582 @@
+    _log("get_data_path")
+
+    dev = xbmc.translatePath( __settings__.getAddonInfo('Profile') )
+    
+    # Parche para XBMC4XBOX
+    if not os.path.exists(dev):
+        os.makedirs(dev)
+
+    _log("get_data_path ->'"+str(dev)+"'")
+
+    return dev
+
+def get_setting(name):
+    _log("get_setting name='"+name+"'")
+
+    dev = __settings__.getSetting( name )
+
+    _log("get_setting ->'"+str(dev)+"'")
+
+    return dev
+
+def set_setting(name,value):
+    _log("set_setting name='"+name+"','"+value+"'")
+
+    __settings__.setSetting( name,value )
+
+def open_settings_dialog():
+    _log("open_settings_dialog")
+
+    __settings__.openSettings()
+
+def get_localized_string(code):
+    _log("get_localized_string code="+str(code))
+
+    dev = __language__(code)
+
+    try:
+        dev = dev.encode("utf-8")
+    except:
+        pass
+
+    _log("get_localized_string ->'"+dev+"'")
+
+    return dev
+
+def keyboard_input(default_text="", title="", hidden=False):
+    _log("keyboard_input default_text='"+default_text+"'")
+
+    keyboard = xbmc.Keyboard(default_text,title,hidden)
+    keyboard.doModal()
+    
+    if (keyboard.isConfirmed()):
+        tecleado = keyboard.getText()
+    else:
+        tecleado = ""
+
+    _log("keyboard_input ->'"+tecleado+"'")
+
+    return tecleado
+
+def message(text1, text2="", text3=""):
+    _log("message text1='"+text1+"', text2='"+text2+"', text3='"+text3+"'")
+
+    if text3=="":
+        xbmcgui.Dialog().ok( text1 , text2 )
+    elif text2=="":
+        xbmcgui.Dialog().ok( "" , text1 )
+    else:
+        xbmcgui.Dialog().ok( text1 , text2 , text3 )
+
+def message_yes_no(text1, text2="", text3=""):
+    _log("message_yes_no text1='"+text1+"', text2='"+text2+"', text3='"+text3+"'")
+
+    if text3=="":
+        yes_pressed = xbmcgui.Dialog().yesno( text1 , text2 )
+    elif text2=="":
+        yes_pressed = xbmcgui.Dialog().yesno( "" , text1 )
+    else:
+        yes_pressed = xbmcgui.Dialog().yesno( text1 , text2 , text3 )
+
+    return yes_pressed
+
+def selector(option_list,title="Select one"):
+    _log("selector title='"+title+"', options="+repr(option_list))
+
+    dia = xbmcgui.Dialog()
+    selection = dia.select(title,option_list)
+
+    return selection
+
+def set_view(view_mode, view_code=0):
+    _log("set_view view_mode='"+view_mode+"', view_code="+str(view_code))
+
+    # Set the content for extended library views if needed
+    if view_mode==MOVIES:
+        _log("set_view content is movies")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"movies" )
+    elif view_mode==TV_SHOWS:
+        _log("set_view content is tvshows")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"tvshows" )
+    elif view_mode==SEASONS:
+        _log("set_view content is seasons")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"seasons" )
+    elif view_mode==EPISODES:
+        _log("set_view content is episodes")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"episodes" )
+
+    # Reads skin name
```

Comment:
Please remove everything in this method below this line. Setting views is not allowed anymore in the repository.